olive/model/__init__.py

```diff
@@ -475,7 +475,8 @@ def __init__(
         model_loader: Union[str, Callable] = None,
         model_script: Union[str, Path] = None,
         script_dir: Union[str, Path] = None,
-        io_config: Union[Dict[str, Any], IOConfig, str] = None,
+        io_config: Union[Dict[str, Any], IOConfig] = None,
+        io_config_func: Union[str, Callable] = None,
```

Comment:
could we unify them into io_config?

Reply:
Yes, we can do it. That is simpler too.

I remembered in a previous discussion, you mentioned that `io_config` name is confusing when it is a func https://github.com/microsoft/Olive/pull/703#discussion_r1385845515, so I made them separate. 

But I think unifying them and making the type `Union[Dict, IOConfig, str, Callable]` makes the logic simpler. Even though io_config gets overloaded with two different types of values: the config itself or a function to load the config.

Reply:
I recommend unifying it to use just `io_config`.

Reply:
Yes, already pushed a commit to unify them. Fixing some bugs/tests now.